From 0ad92f7d174b25bc3c4a7a29184f8665a58cd119 Mon Sep 17 00:00:00 2001
From: Matthieu Gallien <matthieu_gallien@yahoo.fr>
Date: Fri, 13 Dec 2019 09:53:23 +0100
Subject: [PATCH 1/3] use global menu only on linux due to behavior difference
 on Windows

on Windows, the menu is always shown like a traditionnal menu inside the
window. We do not want that.
---
 src/qml/ElisaMainWindow.qml     | 6 ------
 src/qml/PlatformIntegration.qml | 6 ++++++
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/qml/ElisaMainWindow.qml b/src/qml/ElisaMainWindow.qml
index 1151c90..1319a4e 100644
--- a/src/qml/ElisaMainWindow.qml
+++ b/src/qml/ElisaMainWindow.qml
@@ -21,7 +21,6 @@ import QtQuick.Layouts 1.1
 import QtQuick.Window 2.2
 import org.kde.elisa 1.0
 import Qt.labs.settings 1.0
-import Qt.labs.platform 1.0 as NativeMenu
 
 ApplicationWindow {
     id: mainWindow
@@ -79,11 +78,6 @@ ApplicationWindow {
         id: applicationMenu
     }
 
-    NativeMenu.MenuBar {
-        NativeApplicationMenu {
-        }
-    }
-
     SystemPalette {
         id: myPalette
         colorGroup: SystemPalette.Active
diff --git a/src/qml/PlatformIntegration.qml b/src/qml/PlatformIntegration.qml
index cc6ef9b..36f4984 100644
--- a/src/qml/PlatformIntegration.qml
+++ b/src/qml/PlatformIntegration.qml
@@ -16,6 +16,7 @@
  */
 
 import QtQuick 2.7
+import Qt.labs.platform 1.0 as NativeMenu
 import org.kde.elisa 1.0
 
 Item {
@@ -29,6 +30,11 @@ Item {
 
     signal raisePlayer()
 
+    NativeMenu.MenuBar {
+        NativeApplicationMenu {
+        }
+    }
+
     Mpris2 {
         id: mpris2Interface
 
-- 
2.24.0

